helpdesk/models/helpdesk_ticket.py

```diff
@@ -83,6 +88,18 @@ def _onchange_partner_id(self):
             self.partner_name = self.partner_id.name
             self.partner_email = self.partner_id.email
 
+    @api.onchange('team_id')
+    def _onchange_team_id(self):
+        if not self.team_id:
```

Comment:
This `if` will not be necessary because we must allow to pick an user without a team.